frontend/src/edit.jsx

```diff
@@ -1,21 +1,142 @@
+  } = props;
 
+  const [activeTab, setActiveTab] = useState('create');
+  const [scenarioName, setScenarioName] = useState('');
+  const [scenarioID, setScenarioID] = useState(null);
+  const [pattern, setPattern] = useState('');
+  const [parcelTable, setParcelTable] = useState(null);
+  const [jobID, setJobID] = useState(null);
 
-  const { open } = props
-  console.log('render edit menu', open);
+  useEffect(async () => {
+    if (parcelID) {
+      const table = await getLulcTableForParcel(parcelCoords);
+      setParcelTable(table);
+    }
+  }, [parcelID]);
 
-  if (open) {
-    return (
-      <div className={"menu-container"}>
-        <span>Wallpaper with pre-made></span>
-        {/*<Dropdown>
+  useInterval(async () => {
+    console.log('checking status for', jobID);
+    const status = await getStatus(jobID);
+    if (status === 'complete') {
+      const results = await getWallpaperResults(jobID);
+      setParcelTable(results);
+      setJobID(null);
+    }
+  }, jobID ? 1000 : null);
 
-        </Dropdown>*/}
-      </div>
-    );
-  } else {
-    return <div />
+  async function handleSubmitNew(event) {
+    event.preventDefault();
+    if (!scenarioName) {
+      alert('no scenario was selected');
+      return;
+    }
+    if (!pattern) {
+      alert('no modification options selected; no changes to make');
+      return;
+    }
+    if (!parcelID) {
+      alert('no parcel was selected; no changes to make');
+      return;
+    }
+    let sid = scenarioID;
+    if (!Object.values(savedScenarios).includes(scenarioName)) {
+      sid = await makeScenario(scenarioName, 'description');
+      setScenarioID(sid);
+    }
+    const jid = await doWallpaper(parcelCoords, pattern, sid);
+    setJobID(jid);
+    refreshSavedScenarios();
   }
+
+  // TODO: do handlers  need to be wrapped in useCallback? 
+  // to memoize the function?
+  const handleRadio = (event) => {
+    setPattern(event.target.value);
+  };
+
+  const handleTabChange = (tabID) => {
+    setActiveTab(tabID);
+  };
+
+  return (
+    <div className="menu-container">
+      <Tabs id="Tabs" onChange={handleTabChange} selectedTabId={activeTab}>
+        <Tab
+          id="create"
+          title="Create"
+          panel={(
+            <div>
+              <ParcelTable parcelTable={parcelTable} />
+              <br />
+              <form onSubmit={handleSubmitNew}>
+                <RadioGroup
+                  inline={false}
+                  label="Modify the landuse of this parcel by selecting a pattern:"
+                  onChange={handleRadio}
+                  selectedValue={pattern}
+                >
+                  <Radio
+                    value="orchard"
+                    label="orchard"
+                  />
+                  <Radio
+                    value="city park"
+                    label="city park"
+                  />
+                  <Radio
+                    value="housing"
+                    label="housing"
```

Comment:
Yes! This will become the "catalog" of patterns we're talking about. So might end up being a dropdown, or something else. And values will come from the backend, which has a database populated by people using the thing we create in #17